### PR TITLE
Disallow relative module specifies that don't start with "./", "../"

### LINKED ISSUE #1769: Disallow relative module specifies that don't start with "./", "../", "/"
Apparently web browsers don't allow this:
```js
import 'bar.js'
```
Chrome: `Uncaught TypeError: Failed to resolve module specifier "bar.js". Relative references must start with either "/", "./", or "../".`
Firefox: `TypeError: Error resolving module specifier: bar.js`
Safari: `TypeError: Module specifier does not start with "/", "./", or "../".`

Thanks to @thgh [for reporting this](https://github.com/denoland/deno/issues/1750#issuecomment-463129189).

Does anyone have a reference to the standard spec that states this?

We should also error on this. We're currently doing this all over the place.

### PR DESCRIPTION
**Pull request by manyuanrong**
Closes #1769

### CODE REVIEW

File: src/deno_dir.rs [hunk 1]
```diff
@@ -520,6 +536,14 @@ fn is_remote(module_name: &str) -> bool {
...
+  module_name.starts_with("./") || module_name.starts_with("../")
+}
+
+fn is_absolute(module_name: &str) -> bool {
+  module_name.starts_with("/") || module_name.starts_with("file://")
+}
+
 fn parse_local_or_remote(p: &str) -> Result<url::Url, url::ParseError> {
   if is_remote(p) || p.starts_with("file:") {
     Url::parse(p)
```
Comment (ry) [inline src/deno_dir.rs]: Can you add a unit test to the bottom of this file that demos the change?

**Comment by manyuanrong** [comment]
@ry  I submitted some test cases, and the deno_std code has been merged, but I don't know why CI still failed, because it seems that the deno_std/testing code is still the previous one.

**Comment by ry** [comment]
deno_std has to be upgraded manually - which I'm trying to do here: https://github.com/denoland/deno/pull/1833
Looks like this will be fine after that lands.

**Comment by ry** [comment]
@manyuanrong Could you please rebase? I think it should work now.

**Comment by manyuanrong** [comment]
There still seems to be a problem with appveyor, but I'm sorry I don't know why. I tried to find the reason, but unfortunately I failed.

**Comment by zekth** [comment]
> There still seems to be a problem with appveyor, but I'm sorry I don't know why. I tried to find the reason, but unfortunately I failed.

For the error in `appveyor` the only part which changes is the import is a `json` file instead of a `ts` file. I will take a look.

**Comment by ry** [comment]
Thanks for the PR @manyuanrong, this got the ball rolling which eventually resulted in #2084 - solving the issue. There was some infrastructure work to do first, which is why this wasn't merged.

**Pull request closed without merging**